trillian_map_api.proto

```diff
@@ -76,6 +76,30 @@ message GetMapLeavesByRevisionRequest {
   int64 revision = 3;
 }
 
+message ListLeavesByRevisionRequest {
+  int64 map_id = 1;
+  int64 revision = 2;
+  // [start_index, end_index)
+  bytes start_index = 3;
+  bytes end_index = 4;
+  // A comma separated list of fields. The default sorting order is ascending.
+  // To specify decending order for a field, a suffix " desc" should be appended
+  // to the field name. For example: "foor desc,bar".
+  // The only supported field is "index". Valid options are "index", and "index desc".
+  string order_by = 5;
+  // The maximum number of items to return.
+  int32 page_size = 6;
+  // The next_page_token value returned from a previous List request, if any.
+  string page_token = 7;
```

Comment:
I'm a little concerned that order_by/page_size/page_token push this in quite a heavy direction. My intuition is that this generic API is really trying to implement two operations and it might be better to expose them separately. e.g:

getLastInRange - returns a single element and proof it is the last
getAllInRange - sets up a streaming grpc to return all of the responses.

Reply:
I was taking my inspiration from the Google Cloud API Design Patterns doc:
https://cloud.google.com/apis/design/standard_methods#list

Reply:
Splitting them up seems like a good idea. It should also make implementation easier and development faster.